app/models/miq_worker/runner.rb

```diff
@@ -374,6 +375,32 @@ def heartbeat_to_drb
   def heartbeat_to_file(timeout = nil)
     timeout ||= worker_settings[:heartbeat_timeout] || Workers::MiqDefaults.heartbeat_timeout
     File.write(@worker.heartbeat_file, (Time.now.utc + timeout).to_s)
+
+    get_messages.each { |msg, *args| process_message(msg, *args) }
+  end
+
+  def get_messages
+    messages = []
+    now = Time.now.utc
+    @my_last_config_change ||= now
+
+    last_config_change = server_last_change(:last_config_change)
+    if last_config_change && last_config_change > @my_last_config_change
+      _log.info("#{log_prefix} Configuration has changed, New TS: #{last_config_change}, Old TS: #{@my_last_config_change}")
+      messages << ["sync_config"]
+
+      @my_last_config_change = last_config_change
```

Comment:
Should we change this to this maybe?

```ruby
@my_last_config_change = Time.now.utc
```

Reason being, is you do a `>` check in the if, and if you just set it to the same time, then it will just continue calling `sync_config` every time.